p2p/src/test/kotlin/maru/p2p/discovery/MaruDiscoveryServiceTest.kt

```diff
@@ -236,47 +216,39 @@ class MaruDiscoveryServiceTest {
               P2P.Discovery(
                 port = PORT6,
                 bootnodes = listOf(enrString),
-                refreshInterval = 5.seconds,
+                refreshInterval = 10.seconds,
               ),
           ),
         forkIdHashProvider = forkIdHashProvider,
-        metricsSystem = NoOpMetricsSystem(),
       )
 
     try {
       bootnode.start()
       discoveryService2.start()
       discoveryService3.start()
 
-      awaitPeerFound(bootnode, discoveryService2.getLocalNodeRecord().nodeId)
-      awaitPeerFound(discoveryService2, discoveryService3.getLocalNodeRecord().nodeId)
-      awaitPeerFound(discoveryService3, discoveryService2.getLocalNodeRecord().nodeId)
-      awaitPeerFound(bootnode, discoveryService3.getLocalNodeRecord().nodeId)
+      // make sure services are started and bootnodes have been pinged
```

Comment:
But with the injectable timerFactory you can force it without the sleep. That's the whole point of making it injectable